client/my-sites/checkout/src/components/checkout-main-content.tsx

```diff
@@ -508,6 +511,18 @@ export default function CheckoutMainContent( {
 	}
 
 	const nextStepButtonText = translate( 'Continue to payment', { textOnly: true } );
+	const canEditStep = () => {
+		if ( ! paymentMethods ) {
+			return false;
+		}
+		const containsFreeOrCreditMethod = paymentMethods.filter(
```

Comment:
Super minor but this can use `some` instead of `filter`. If it was a long list it would be slightly more efficient.

```suggestion
		const containsFreeOrCreditMethod = paymentMethods.some(
```